### PR TITLE
build(deps): migrate kubernetes provider to >= 2.0.0

### PR DESCRIPTION
**Pull request by sbeginCoveo**
This PR migrates the kubernetes provider to >= 2.0.0 in order to be used in terraform projects which require this version of the provider. Any suggestions on version semantics and documenting/changelog are welcome

## Breaking changes
* Upgraded Kubernetes provider to >= 2.0.0. This means a migration of the host terraform project is required, to use this same provider version. Heres the [upgrade guide](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/guides/v2-upgrade-guide)

### CODE REVIEW

**Comment by alannix-lw** [comment]
@afiune I tested a similar change when we released this module - the v2 Kubernetes provider was brand new at that point.  This is a breaking change for anyone who relies on v1 of the Kubernetes provider, so we just need to document and version as such.

**Comment by afiune** [comment]
Closing and opening this PR to launch our CI tests.

**Comment by afiune** [comment]
First of all, @sbeginCoveo thank you for your contribution!! 🎉 

We are going to test this internally, as @alannix-lw mentioned, this will be a breaking change and therefore, we will have to do a major version bump to `1.0.0`. We handle our release process with automation (the changelog and release notes generation).

I will add this content to the PR description:
## Breaking changes
* Upgraded Kubernetes provider to >= 2.0.0. This means a migration of the host terraform project is required, to use this same provider version. Heres the [upgrade guide](https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/guides/v2-upgrade-guide)

**Review by scottford-lw — APPROVED** [review]

**Comment by sbeginCoveo** [comment]
Sorry for the delay. Please tell me if anything's missing :)

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.3.1-dev
\ No newline at end of file
+1.0.0
```
Comment (afiune) [inline VERSION:1]: Very nice that you bumped the major version, though, we remove the `-dev` tag only when we are doing a release.
```suggestion
1.0.0-dev
```

**Review by afiune — APPROVED** [review]
One little thing to change and we are ready to roll!

tenor-162288238

File: VERSION (hunk 1 above)
Comment (sbeginCoveo) [reply VERSION]: :heavy_check_mark:

**Comment by sbeginCoveo** [comment]
:wave:

**Review by afiune — APPROVED** [review]
tenor-75360197

**Pull request merged**